src/Tasks/CVECheckTask.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+use SensioLabs\Security\SecurityChecker;
+
+/**
+ * Checks if there are any insecure dependencies.
+ */
+class CVECheckTask extends BuildTask
+{
+    /**
+     * @var SecurityChecker
+     */
+    protected $securityChecker;
+
+    private static $dependencies = [
+        'SecurityChecker' => '%$' . SecurityChecker::class,
+    ];
+
+    protected $title = 'Composer security checker';
+
+    protected $description =
+        'Checks if any modules managed through composer have known security vulnerabilities at the used version.';
+
+    /**
+     * @return SecurityChecker
+     */
+    public function getSecurityChecker()
+    {
+        return $this->securityChecker;
+    }
+
+    /**
+     * @param SecurityChecker $securityChecker
+     * @return CVECheckTask $this
+     */
+    public function setSecurityChecker($securityChecker)
+    {
+        $this->securityChecker = $securityChecker;
+        return $this;
+    }
+
+    public function run($request)
+    {
+        // to keep the list up to date while removing resolved issues we keep all of found issues
+        $validEntries = array();
+
+        // use the security checker of
+        $checker = $this->getSecurityChecker();
+        $alerts = $checker->check(BASE_PATH . DIRECTORY_SEPARATOR . 'composer.lock');
```

Comment:
I feel like we should use the ComposerLoader from silverstripe-maintenance for something here, at the least to remove the reliance on the physical path of composer.lock from this code (and leave it to ComposerLoader)

Reply:
1) This library has it's own logic for reading and parsing the lock file, and only requires the path in the check method.
2) ComposerLoader requires a base path, which means either specifying manually still, or relying on `BASE_PATH` as it is here.
3) There is no way at current to extract the lock path from ComposerLoader (only the parsed contents).
4) This modules does not require silverstripe-maintenance, nor is it intended to.
5) This task is for checking the main installation only, not another at some arbitrary location, nor does it provide the ability to upload and check random lock files.

**Conclusion:**
ComposerLoader should not be used for this. Although I could add a setter/getter for a lock path, defaulting to what is currently set, I feel it is unnecessary.